conan/tools/cmake/cmake.py

```diff
@@ -57,38 +58,48 @@ def __init__(self, conanfile):
 
         self._generator = configure_preset["generator"]
         self._toolchain_file = configure_preset.get("toolchainFile")
-        self._cache_variables = configure_preset["cacheVariables"]
-
+        self._presets_cache_variables = configure_preset["cacheVariables"]
         self._cmake_program = "cmake"  # Path to CMake should be handled by environment
 
+    @property
+    def _cache_variables(self):
+        """
+        Get all the variables that should be passed the CLI as ``cmake -Dxxxxx -Dyyyyy ....``
+        """
+        cache_variables = {}
+        generator_folder = self._conanfile.generators_folder
+        if self._toolchain_file:
+            if os.path.isabs(self._toolchain_file):
+                toolchain_file = self._toolchain_file
+            else:
+                toolchain_file = os.path.join(generator_folder, self._toolchain_file)
```

Comment:
I know this is not from your PR, but I think this might be unnecessary, the toolchain file should be always there or we probably have to raise an error, and about the absolute path, we always write the absolute path unless I'm not seeing something. 
I say this because I would like to simplify so the toolchain can be directly extracted from the `CMakePresets.json` file.
And why? well, I still see the "initial cache" isolated from this. I would mind if the logic from there is reading also the toolchain from the file and not reusing code from the build helper.